mcs/class/monodoc/Test/Monodoc.Ecma/EcmaUrlTests.cs

```diff
@@ -454,38 +470,293 @@ public void ExplicitIndexerImplementation ()
 			AssertUrlDesc (ast, "P:System.Web.SessionState.HttpSessionStateContainer$System.Web.SessionState.IHttpSessionState.Item(System.Int32)");
 		}
 
-		/*		[Test]
-		public void TreeParsabilityTest ()
+		[Test]
+		public void ToEcmaCref_Namespace ()
 		{
-			var rootTree = RootTree.LoadTree ("/home/jeremie/monodoc/");
-			Node result;
-			var generator = new CheckGenerator ();
+			var actual = new EcmaDesc {
+				DescKind = EcmaDesc.Kind.Namespace,
+				Namespace = "System.IO"
+			};
 
-			foreach (var leaf in GetLeaves (rootTree.RootNode).Where (IsEcmaNode))
-				AssertUrl (leaf.PublicUrl);
+			AssertEcmaString ("N:System.IO", actual);
 		}
 
-		IEnumerable<Node> GetLeaves (Node node)
+		[Test]
+		public void ToEcmaCref_SimpleType ()
 		{
-			if (node == null)
-				yield break;
+			var actual = new EcmaDesc {
+				DescKind = EcmaDesc.Kind.Type,
```

Comment:
it is public: https://github.com/mono/mono/blob/master/mcs/class/monodoc/Monodoc.Ecma/EcmaDesc.cs#L13